### ISSUE TITLE
Define folder structure and also modify k8 manifest files to deploy in named namespaces 

### ISSUE THREAD

**Issue by arvindclement** [tranquilitybase-io/tb-gcp#329]
## Purpose
Applications that get deployed on the shared gke cluster are all accessing one big k8 manifest file called eagle_console.yaml becoming cumbersome to manage as we need to add in Jenkins and the DAC deployments as well.

## Estimated Work to Complete
Steps to complete the task
1. under the /opt/tb/repo/tb-gcp-tr/landingZone folder create a folder called shared-dac
2. Add yaml file to create 2 namespaces ssp(self service portal) and cicd
3. Add separate yaml files with code for eagle_console, houston service and jenkins for e.g files called eagle_console.yaml, houston_service.yaml, jenkins.yaml
3. Modify the terraform code in landingZone/no-itop to deploy from the shared-dac folder 

## Definition of Done
when you run the landing zone deploy the apps get deployed onto the shared gke cluster in the relevant namespaces

## Related Issues
links to other issues

**Comment by andy12838729** [comment]
We need to create a Shared Services folder - this shouldnt be in the LZ Folder

**Comment by arvindclement** [comment]
the main.tf file the order in which the services are started matters at the moment as the kube.sh file gets filled according and the yaml are executed in order as well so use the depends on in the module to enfore ordering

**Comment by arvindclement** [comment]
i have sorted the issue with the k8 yaml file not getting executed as part of a deploy the issue is with the exit 0,1 statements and also the order in which the kubectl apply -f commands get put into the kube.sh file. This is where the depends on clause in the module is used to determine the order the istio stuff needs to be at the very end.

**Comment by arvindclement** [comment]
There are 2 new folders created in the tb-gcp-tr shared-frontend and shared-devops 
The share-frontend folder will contains the following files
1.) namespace.yaml which would contains the definition for creating the ssp namespace
2.) eagle_console.yaml file for deploying eagle console
3.) houston_service.yaml file for deploying houston service
the shared share-devops folder would have the following yaml files
1.) namspace.yaml file to define the scicd namespace
2.) the dac.yaml for deploying dac
3.) jenkins.yaml file for deploying jenkins

**Comment by arvindclement** [comment]
The changes have been done in branch issue-329 going to test it tomorrow also @wills-gft needs to put in the Jenkins file and then we can test a deploy

**Comment by arvindclement** [comment]
- So the folder outside landingZone is going to be called shared-dac.

- Test having a namespace.yaml and made modification to the landing zone deploy to deploy this into the gke cluster.

- Next is to create separate manifest files for eagle_console, houston service(include mysql database), dac and jenkins.yaml


inside the shared-dac there is going to be a folder called eagle-console and there would be the eagle console yaml file and the houston service

**Comment by arvindclement** [comment]
Istio needs to be deployed in the ssp and cicd namespaces

**Comment by arvindclement** [comment]
successfully tested the deploy of eagle console, houston service and mysql database with istio service mesh in place on relevant namespaces.  The only thing which is pending is istio to be enabled on the Jenkins namespace which Caron will provide the details once that is done shall test and bundle the changes into a PR

**Comment by arvindclement** [comment]
Had to put this on hold for the DB demo that happened on the 17/06/2008. Also as part of the learning that came up from that demo we would put the DAC in as a single file eagle console file and then look at separating the services

**Comment by arvindclement** [comment]
As this task became too big has to break it down to the follow #279 has the follow

1. New folder called shared-dac under /opt/tb/repo/tb-gcp-tr/landingZone 
2. Yaml file in the shared-dac folder to create 2 namespaces ssp(self service portal) and cicd
3. Yaml file in the shared-dac folder to have the jenkins-master running
4. Modify the terraform code in landingZone/no-itop to deploy namespace and the jenkins from the shared-dac folder

going to create a separate ticket to handle the split of eagle console, house service and dac into separate yaml files and also create them on separate ssp namespace
